rotkehlchen/chain/ethereum/uniswap/typing.py

```diff
@@ -65,10 +77,18 @@ def serialize(self) -> Dict[str, Any]:
 
 
 AddressBalances = Dict[ChecksumEthAddress, List[LiquidityPool]]
+DDAddressBalances = DefaultDict[ChecksumEthAddress, List[LiquidityPool]]
 AssetPrice = Dict[ChecksumEthAddress, Price]
 
 
 class ProtocolBalance(NamedTuple):
     address_balances: AddressBalances
     known_assets: Set[EthereumToken]
     unknown_assets: Set[UnknownEthereumToken]
+
+
+# Get History
+
+AddressTrades = Dict[ChecksumEthAddress, List[AMMTrade]]
+DDAddressTrades = DefaultDict[ChecksumEthAddress, List[AMMTrade]]
+ProtocolHistory = Dict[str, Union[AddressTrades]]
```

Comment:
Dicts as types are very hard to work with. mypy can't process them. Unless we use typed dict which we don't. I generally prefer to use named tuples. It's much clearer and easier to pass around. And as an extra bonus mypy typing works much better with it